src/modules/generateRandomNumber.js

```diff
@@ -7,7 +7,21 @@
  * @return {number} A random 4-digit number
  */
 function generateRandomNumber() {
-  /* Write your code here */
+  let digits = ['1', '2', '3', '4', '5', '6', '7', '8', '9'];
+  let number = digits[Math.floor(Math.random() * digits.length)];
+
+  digits = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'].filter(
+    (d) => !number.includes(d),
+  );
+
+  while (number.length < 4) {
+    const randomIndex = Math.floor(Math.random() * digits.length);
+
+    number += digits[randomIndex];
+    digits.splice(randomIndex, 1);
+  }
+
+  return Number(number);
```

Comment:
The function returns the number as a `Number` type. If the first randomly generated digit is '0' after modifying the function to include '0' as a possible first digit, `Number(number)` will remove it, which may not be the desired outcome if the leading zero is significant. Consider returning the number as a string if leading zeroes are important.